CMakeLists.txt

```diff
@@ -472,8 +472,11 @@ endif()
 option(STATIC "Link libraries statically" ${DEFAULT_STATIC})
 
 # This is a CMake built-in switch that concerns internal libraries
-if (NOT DEFINED BUILD_SHARED_LIBS AND NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
-    set(BUILD_SHARED_LIBS ON)
+set(BUILD_SHARED_LIBS_HELP_TEXT "Build internal libraries as shared")
+if (NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+    option(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS_HELP_TEXT} ON)  # ON for debug, otherwise the build consumes way too much disk space.
```

Comment:
Correct me if I'm wrong but it's not on for static debug, I would edit the comment otherwise the code and the comment don't match.